src/xpath.c

```diff
@@ -4076,14 +4085,16 @@ xpath_deref(struct lyxp_set **args, uint32_t UNUSED(arg_count), struct lyxp_set
                             lyd_get_value(&leaf->node));
                     return LY_EVALID;
```

Comment:
Once you define `cleanup`, it is best to always go there even when not needed (because it might later and the expectation is that cleanup code is always executed).

Reply:
Done